File: Visitor/PHP/DebitCard.php
<?php

require_once "IPayment.php";
require_once "CPU.php";
require_once "VGA.php";

class DebitCard implements IPayment {
    public function visit($component) {
        if ($component instanceof CPU) {
            echo "Purchase CPU with Debit Card\n";
        } elseif ($component instanceof VGA) {
            echo "Purchase VGA with Debit Card\n";
        }
    }
}

File: Visitor/Dart/DebitCard.dart
import 'CPU.dart';
import 'IPayment.dart';
import 'VGA.dart';

class DebitCard implements IPayment {
  @override
  void visit(component) {
    if (component is CPU) {
      print("Purchase CPU with Debit Card");
    } else if (component is VGA) {
      print("Purchase VGA with Debit Card");
    }
  }
}
